src/data/actions/assets.js

```diff
@@ -1,31 +1,47 @@
 import * as clientApi from '../api/client';
 import { assetActions } from '../constants/actionTypes';
 
+const isCorrectResponse = (request, lastRequest) => (
```

Comment:
This function verifies whether the current state of lastRequest (the request currently stored in Redux) is equal to the request that was made in the currently executing getAssets call. This accounts for cases where multiple requests are made and the underlying Redux request state may change from underneath an API call and result in inconsistent state in the UI.

An example is clicking all the filters, unclicking each filter one by one (from Other up to Audio) and ending up with only Audio files displayed (instead of all files). This happens because the API call that is made last (all documents) comes back long before the second to last API call (audio).

Reply:
I agree. I think Dennis and I had talked about this being a better option, but I don't know how widely supported that is, per [this](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) page. It sounds like not all browsers support aborting a fetch ('experimentally' also doesn't inspire much confidence...), so I'm not sure how tractable that will be if we're using fetch?

Reply:
Wow, I'm surprised there's no way to cancel a request right now... seems like a pretty basic feature. It looks like that [the fetch pollyfill](https://github.com/edx/studio-frontend/blob/master/package.json#L30) that we are using is going to add support for aborts soon though: https://github.com/github/fetch/pull/592.

Reply:
If we really wanted to implement this we should consider dropping fetch for straight up XHR requests because they support cancelling ([but there might be some other drawbacks to them](https://stackoverflow.com/a/35552420/6620612)). I think [axios](https://github.com/axios/axios) is a good XHR client library that we could use.